math-lib/math/private/flonum/flonum-functions.rkt

```diff
@@ -133,6 +133,7 @@
     (let ([xa  (flmin xa ya)]
           [ya  (flmax xa ya)])
       (cond [(fl= xa 0.0)  ya]
+            [(flinfinite? xa) xa]
```

Comment:
This works, but why test `xa` instead of `ya`? If either argument is infinite, that's enough to conclude the result is infinite (unless either is NaN, but in that case _both_ `xa` and `ya` are NaN).